### ISSUE TITLE
unusual_byte_groupings: ...u64 trailing type is included in grouping calculation

### ISSUE THREAD

**Issue by cyphar** [rust-lang/rust-clippy#15210]
### Summary

The `unusual_byte_groupings` lint appears to treat trailing `u64` and similar type casts as being part of the digit grouping, resulting in very unnatural suggestions for better groupings.

(In my example I'm using a `const` where it's trivial to work around the issue, but in my actual codebase this is in a `#[repr(u64)] enum` which needs these type hints in order for cbindgen to produce reasonable constant values.)

### Lint Name

unusual_byte_groupings

### Reproducer

I tried this code:

```rust
const A = 0xFFFF_FFFE_3EAD_5E1Fu64;
```

I saw this happen:

```
error: digits of hex, binary or octal literal not in groups of equal size
   --> example.rs:101:31
    |
101 |     const A = 0xFFFF_FFFE_3EAD_5E1Fu64;
    |               ^^^^^^^^^^^^^^^^^^^^^^^^ help: consider: `0x0FFF_FFFF_E3EA_D5E1_Fu64`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unusual_byte_groupings
```

I expected to see this happen:

No lint warning.

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: x86_64-unknown-linux-gnu
release: 1.88.0
LLVM version: 20.1.5
```

**Comment by cyphar** [comment]
Ah, it seems this actually only happens with `#[open_enum]`-tagged enums... I guess this might be an issue with proc-macro enums?
